### ISSUE TITLE
Long runtime - Optimization and timeout needed

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/metalearn#15]
Metafeatures take a long time to compute when the dataset is not small. On a dataset of about 500k instances with about 60 features or about 90MB, computing metafeatures is taking on the order of 10s of minutes. This example is not really big data.

First, computing metafeatures should be optimized.

Second, perhaps a timeout feature could be implemented with priority given based on user input (#10) or based on some default priority, like quickest to compute first.

What can be done to make computing metafeatures more reasonable?

**Comment by bjschoenfeld** [comment]
Optimization is not a clear todo here. We have done some and there will always be more to do. If a particular optimization presents itself, let's create a new issue for that item.

Let's focus on getting the timeout feature. If the user gives a a max time, we will return all the metafeatures computed by that time. If the user specified particular metafeatures, then we should compute them in that order, otherwise whatever default we have now is great.

**Comment by bjschoenfeld** [comment]
I think the timeout should be for the entire compute function, not just computing metafeatures. This includes input validation and data pre-processing. This will allow the user to reliably and safely timeout the call to compute metafeatures.

**Comment by bjschoenfeld** [comment]
I added a two second buffer to the input timeout, so that the compute function can use those two seconds for computational overhead without going over time.

**Comment by bjschoenfeld** [comment]
I have tried using signal.alarm to timeout the compute function. It seems unreliable in a multi-process environment. I am not sure if the unit tests are in parallel. I believe that some of the metafeatures are computed using packages which use parallel processing. How can we reliably timeout the compute function and reliably test it?
